### PR TITLE
Ensure decaton works with older jdk

### PR DESCRIPTION
**Pull request by ocadaruma**
- Now Decaton compiles only on jdk >= 21 (by https://github.com/line/decaton/pull/224) but since there are many users who still use jdk 8,11,17, we still should run tests on these versions

### CODE REVIEW

File: client/src/test/java/com/linecorp/decaton/client/DecatonClientTest.java [hunk 1]
```diff
@@ -39,8 +39,7 @@
 
 @ExtendWith(MockitoExtension.class)
 public class DecatonClientTest {
-    @Spy
-    private final DecatonClient<HelloTask> decaton = new DecatonClient<HelloTask>() {
+    private static class NoopClient implements DecatonClient<HelloTask> {
```
Comment (ocadaruma) [inline client/src/test/java/com/linecorp/decaton/client/DecatonClientTest.java:42]: Spying anonymous class fails on JDK8 with reflection error
https://github.com/line/decaton/actions/runs/8174863099/job/22350646261
image

I didn't want to dig into the detail as we likely drop JDK8-support in not-so-far-future so I just rewrote it with static class for now.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,14 +12,23 @@ jobs:
...
     - uses: actions/checkout@v2
     - name: Setup java
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v4
       with:
-        java-version: ${{ matrix.java }}
+        distribution: temurin
+        java-version: |
+          ${{ matrix.java }}
+          21
```
Comment (kawamuray) [inline .github/workflows/ci.yml:24]: Is this intentionally left or just forgot to remove line?

File: processor/src/it/java/com/linecorp/decaton/processor/VThreadCoreFunctionalityTest.java [hunk 3]
```diff
@@ -33,6 +35,7 @@
 import com.linecorp.decaton.testing.RandomExtension;
 import com.linecorp.decaton.testing.processor.ProcessorTestSuite;
 
+@EnabledForJreRange(min = JRE.JAVA_21)
```
Comment (kawamuray) [inline processor/src/it/java/com/linecorp/decaton/processor/VThreadCoreFunctionalityTest.java:38]: Oh... there's such a convenient annotation existed... !

**Review by kawamuray — CHANGES_REQUESTED** [review]
just one minor question

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ocadaruma) [reply .github/workflows/ci.yml:24]: Intentionally. We need to install 21 always to build the project, regardless test-jdk version.
Last one will be used as the global java version (refs: https://github.com/actions/setup-java?tab=readme-ov-file#install-multiple-jdks)

Comment (kawamuray) [reply .github/workflows/ci.yml:24]: ah-ok. now understand how it works

**Review by kawamuray — APPROVED** [review]
LGTM, thanks for the quick follow-up!

**Pull request merged**